### DISCUSSION TITLE
How to trigger ROOT_EFFECTS_INIT event in ngrx unit test

### DISCUSSION THREAD

**Discussion by tirelyl** [ngrx/platform#3626] [Q&A] [answered]
I have two exactly the same code to detect whether the language initialization method is called when the project initializes

But `ROOT_EFFECTS_INIT` can't work properly like my custom action `INIT` . It seems that 

![image](https://user-images.githubusercontent.com/20921324/197158373-dbaf10a8-0dd5-4e93-b829-9cefb56bf82c.png)

<img width="736" alt="image" src="https://user-images.githubusercontent.com/20921324/197158560-fa0924ef-8b8a-4a2c-b800-3887e6b5dce1.png">

It looks like ROOT_EFFECTS_INIT cannot be called in unit tests

**Comment by timdeschryver** [comment, marked as answer]
That's because `ROOT_EFFECTS_INIT` is just a string, you need to create an action of it so that the effect picks it up.

For example with

```ts
actions$ = of({type: ROOT_EFFECTS_INIT});
``` 

Or, you can use `rootEffectsInit` which is the action with `ROOT_EFFECTS_INIT` as its type.

**Reply by tirelyl** [reply]
thanks very much！
